pkg/plugins/runtime/gateway/kumactl/plugin.go

```diff
@@ -27,21 +28,12 @@ func (p *Plugin) CustomizeContext(root *kumactl_cmd.RootContext) {
 			Path: "gateways",
 		},
 	)
-
-	root.TypeArgs["gateway"] = core_mesh.GatewayType
+	entities.All = append(entities.All, entities.Definition{
+		Singular:     "gateway",
+		Plural:       "gateways",
+		ResourceType: core_mesh.GatewayType,
+	})
 }
 
-func (p *Plugin) CustomizeCommand(root *cobra.Command) {
-	get := kumactl_cmd.FindSubCommand(root, "get")
-
-	get.AddCommand(
-		kumactl_get.WithPaginationArgs(
-			kumactl_get.NewGetResourcesCmd(p.rootContext, "gateways", core_mesh.GatewayType, kumactl_get.BasicResourceTablePrinter),
-			&p.rootContext.ListContext,
-		),
-	)
-
-	get.AddCommand(
-		kumactl_get.NewGetResourceCmd(p.rootContext, "gateway", core_mesh.GatewayType, kumactl_get.BasicResourceTablePrinter),
-	)
+func (p *Plugin) CustomizeCommand(_ *cobra.Command) {
```

Comment:
Let's file an issue to track removing this plugin mechanism. I don't think we will need it once you get through.

Reply:
https://github.com/kumahq/kuma/issues/2503